src/snowflake/snowpark/_internal/utils.py

```diff
@@ -1939,6 +1941,151 @@ def remove_comments(sql_query: str, uuids: List[str]) -> str:
+
+    time_travel_mode: str  # 'at' or 'before'
+    statement: Optional[str] = None
+    offset: Optional[int] = None
+    timestamp: Optional[str] = None
+    timestamp_type: Optional[str] = None
+    stream: Optional[str] = None
+
+    @staticmethod
+    def validate_and_normalize_params(
+        time_travel_mode: Optional[Literal["at", "before"]] = None,
+        statement: Optional[str] = None,
+        offset: Optional[int] = None,
+        timestamp: Optional[Union[str, datetime.datetime]] = None,
+        timestamp_type: Optional[Union[str, "TimestampTimeZone"]] = None,
+        stream: Optional[str] = None,
+    ) -> Optional["TimeTravelConfig"]:
+        """
+        Validates and normalizes time travel parameters.
+
+        timestamp_type handling:
+            - datetime with timezone + timestamp_type NOT provided → auto-sets to 'TZ'
+            - datetime with timezone + timestamp_type explicitly provided → uses provided timestamp_type
+            - datetime without timezone (naive) + timestamp_type NOT provided → no casting, raw string (timestamp_type=None)
+            - datetime without timezone (naive) + timestamp_type explicitly provided → uses provided timestamp_type
+            - string timestamps + timestamp_type NOT provided → no casting, raw string (timestamp_type=None, Snowflake native handling)
+            - string timestamps + timestamp_type explicitly provided → uses TO_TIMESTAMP_XXX casting
+
+        This behavior preserves timezone information for datetime objects, provides sensible defaults
+        for naive datetime objects, and lets Snowflake handle string timestamp formats natively unless
+        explicit casting is requested.
+
+        Returns:
+            TimeTravelConfig if time travel is specified, None otherwise.
+        Raises:
+            ValueError: If parameters are invalid.
+        """
+        time_travel_arg_count = sum(
+            arg is not None for arg in (statement, offset, timestamp, stream)
+        )
+
+        # Validate mode
+        if time_travel_mode is None:
+            if time_travel_arg_count == 0:
+                return None
+            else:
+                raise ValueError(
+                    "Must specify time travel mode 'at' or 'before' if any other time travel parameter is provided."
+                )
+
+        if time_travel_mode.lower() not in ["at", "before"]:
+            raise ValueError(
+                f"Invalid time travel mode: {time_travel_mode}. Must be 'at' or 'before'."
+            )
+
+        # Validate exactly one parameter is provided
+        if time_travel_arg_count != 1:
+            raise ValueError(
+                "Exactly one of 'statement', 'offset', 'timestamp', or 'stream' must be provided."
+            )
+
+        # Normalize timestamp
+        normalized_timestamp = None
+        if timestamp is not None:
+            normalized_timestamp = _normalize_timestamp(timestamp)
```

Comment:
nit: do we need _normalize_timestamp any more?
I think the logic can be simplified as str(timestamp).strip() which works for both datetime and str